service/controller/app/v1/resource/chart/desired.go

```diff
@@ -24,6 +24,15 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	if cc.AppCatalog.Name == "" {
+		return &v1alpha1.Chart{
+			ObjectMeta: metav1.ObjectMeta{
```

Comment:
returning empty object but having an object name on it to trigger a deletion event.

Reply:
The desired state should not be conditional in order to cause a different outcome. IMO the desired state should always be the same. The decision on when to delete something should be made when computing the delete patch.

Reply:
When an operator calculates the desired state, it needs to have cc.AppCatalog instance to make configMap, secret and charts. I was thinking about make related information as an empty string, (e.g. tarball) but that would affect creation events as well. 

@xh3b4sd Any suggestion on that?

Reply:
When the desired state cannot be calculated the resource cannot work and should cancel until it is able to compute the desired state again. The desired state defines how the state should be. The current state defines how the state is right now. Deciding under which conditions to apply the desired state should be done when computing the create/update/delete patches. When we have conditionals when computing the desired state, the code is wrong.

Reply:
Maybe you already saw an issue that I linked in the description,  app-operator could not find appCatalog which is the part of helm chart release that deleted earlier. 
@xh3b4sd Based on what you said, I think it's bad design to put `appCatalog` in same helm chart, since we could not figure it out desired state. What do you think? 

BTW, I don't get that why we need to calculate full desired state in a deletion event.